### PR TITLE
Stuck Loading View

### PR DESCRIPTION
**Pull request by Ashod**
- **wsd: trace logging**
- **wsd: reduce logging in WebSocketSession::performWrites()**
- **wsd: better handle view loading while destroying**
- **Unit test to reproduce hang when launching a new document.**

### CODE REVIEW

**Comment by Ashod** [comment]
> Looks sensible to me - thanks Ash; not entirely sure this is the same issue as I'm seeing - in the Kit process. Can you try my WIP unit test against yours and see if it provokes different behavior - I really needed two sessions live before disconnecting in that order to catch my problem which could be another similar race and worth looking at while you're around here.

Sure, will report back.

**Comment by caolanm** [comment]
"Test unit-join-disconnect FAILED. Finished in 75s."

**Comment by Ashod** [comment]
A number of edge-cases were reproduced and fixed with the tests included. Took a bit of effort to deal with them in a reliable manner.

File: test/UnitLoad.cpp [hunk 1]
```diff
@@ -262,9 +263,78 @@ UnitBase::TestResult UnitLoad::testLoadUserPrivateInfo()
...
+        LOK_ASSERT_MESSAGE("cannot get a 2nd active view " + documentURL,
+                           helpers::isDocumentLoaded(socket, testname, true));
+
+        // Now we should have two active views ...
+        TST_LOG("We now have two fully-loaded views");
+
+        std::shared_ptr<http::WebSocketSession> socketslowload =
+            helpers::connectLOKit(socketPoll, uri, documentURL, testname);
+
+        // FIXME: do we get this ? ...
```
Comment (caolanm) [inline test/UnitLoad.cpp:298]: Do we know what needs to happen to fix/drop the FIXMEs?
